tests/template_tests/test_base.py

```diff
@@ -1,8 +1,50 @@
 from django.template import Context, Template, Variable, VariableDoesNotExist
+from django.template.base import DebugLexer, Lexer, TokenType
 from django.test import SimpleTestCase
 from django.utils.translation import gettext_lazy
 
 
+class LexerTests(SimpleTestCase):
+    template_string = '''
+        <html><body>
+        {% if test %}<h1>{{ varvalue }}</h1>{% endif %}
+        </body></html>
+        {#comment {{not a var}} %{not a block}% #}
+        '''
```

Comment:
Since you really just need to check that the three types of tags are included, with their start and end delimiters, I would choose a much more minimal example, and one that doesn't have extraneous whitespace. In a minimal example, each start and end delimiter would occur only once in the expected value.